tests/HolidayDetectorTest.php

```diff
@@ -130,13 +150,10 @@ public function testShowsErrorForInvalidData()
         $invalidFile = realpath(__DIR__.'/../data/holidays') . '/invalid.json';
         file_put_contents($invalidFile, 'invalid JSON');
 
-        try {
-            new HolidayDetector('invalid');
-            $this->fail('Did not throw an error with invalid JSON.');
-        } catch (\RuntimeException $e) {
-            unlink($invalidFile);
-            $this->assertEquals("Invalid holiday data for 'invalid'.", $e->getMessage());
-        }
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage("Invalid holiday data for 'invalid'.");
```

Comment:
I replace the `try...catch` expected exception approach with the `expectException` approach.
And I think this should be better than previous approach.